docs/spec/spec.md

```diff
@@ -0,0 +1,99 @@
+# Specification: Ballerina Log Library
+
+_Owners_: @daneshk @MadhukaHarith92  
+_Reviewers_: @daneshk  
+_Created_: 2021/11/15  
+_Updated_: 2021/11/15  
+_Issue_: [#2347](https://github.com/ballerina-platform/ballerina-standard-library/issues/2347)
+
+# Introduction
+The Log library is used to log information when running applications. It is part of Ballerina Standard Library. [Ballerina programming language](https://ballerina.io/) is an open-source programming language for the cloud that makes it easier to use, combine, and create network services.
+
+# Contents
+
+1. [Overview](#1-overview)
+2. [Logging](#2-logging)
+3. [Logging with Context](#3-logging-with-context)
+4. [Configure Logging](#4-configure-logging)
+5. [Writing Logs to a File](#5-writing-logs-to-a-file)
+
+## 1. Overview
+This specification elaborates on the functionalities available in the Log library.
+
+## 2. Logging
+The Ballerina log API provides functions to log at four levels, which are `DEBUG`, `ERROR`, `INFO`, and `WARN`.
+```ballerina
+log:printDebug("debug log");
+log:printError("error log");
+log:printInfo("info log");
+log:printWarn("warn log");
+```
+
+Users can pass any number of key/value pairs, which need to be displayed in the log message. These can be of the `anydata` type including `int`, `string`, and `boolean`.
+```ballerina
+log:printInfo("info log", id = 845315, name = "foo", successful = true);
+```
+
+Users can also pass an `error:StackFrame[]` as a key/value pair.
+```ballerina
+error e = error("bad sad");
+log:printError("error log", stackTrace = e.stackTrace(), username = "Alex92", id = 845315);
+```
+
+Optionally, an error can be passed to the functions.
+```ballerina
+error e = error("something went wrong!");
+log:printError("error log with cause", 'error = e, id = 845315, name = "foo");
+```
+
+## 3. Logging with Context
```

Comment:
IMO, this is not a separate section. should be a part of logging

Reply:
Fixed in fe465f5